### PR TITLE
Ignore brew dependency installation to fix breakages on MacOS CI.

### PR DESCRIPTION
**Pull request by fruffy**

### CODE REVIEW

File: tools/install_mac_deps.sh [hunk 1]
```diff
@@ -41,12 +42,14 @@ BOOST_LIB="boost@1.85"
...
     autoconf automake ccache cmake libtool
     openssl pkg-config coreutils bison grep ninja
-    ${BOOST_LIB}
 )
 for package in "${REQUIRED_PACKAGES[@]}"; do
   brew_install ${package}
 done
 
+# The boost installation is flaky, do not fail if it fails.
+brew_install ${BOOST_LIB} || echo "Failed to install ${BOOST_LIB}"
```
Comment (vlstill) [inline tools/install_mac_deps.sh:51]: But how does the build work without boost?

Comment (fruffy) [reply tools/install_mac_deps.sh:51]: The problem is that boost is installed fine but the installation of a dependency of that brew package fails (Python 3.12). Unclear to me how to fix that. Maybe there is a flag for that problem.

**Review by asl — APPROVED** [review]
Looks reasonable, but.. meh :(

**Comment by fruffy** [comment]
> Looks reasonable, but.. meh :(

https://github.com/p4lang/p4c/pull/4663 will bring salvation, I am telling ya.

We could also look into a different build manager for MacOS. Or just install boost from scratch, but not a big fan of that..

**Pull request merged**